src/components/modals/authentification/ModalAuthInscription.tsx

```diff
@@ -0,0 +1,191 @@
+/**
+ * Modal avec formulaire d'inscription
+ */
+import { useForm } from 'react-hook-form';
+import { Checkbox } from '@/components/ui/checkbox';
+import { firebaseCreateUser } from '@/api/authentification';
+import { FirestoreCreateDocument } from '@/api/firestore';
+
+type LoginFormTypeInscription = {
+  login: 'string';
+  email: 'string';
+  password: 'string';
+};
+
+type Props = {
+  setContenuModal: React.Dispatch<
+    React.SetStateAction<
+      'init' | 'connexion' | 'inscription' | 'password-forget'
+    >
+  >;
+};
+
+const ModalAuthInscription = ({ setContenuModal }: Props) => {
+  const { handleSubmit, register, setError, reset } =
+    useForm<LoginFormTypeInscription>();
+
+  /**
+   * 3. Creation d'une collection dans la db Firestore
+   * await le retour du Firebase avec les infos users
+   * Copie des infos users utilent dans un objet
+   * Envoi de l'objet dans la db Firestore
+   */
+  const handleCreateUserAuthentification = async (
+    collectionName: string,
+    documentId: string,
+    document: object,
+  ) => {
+    const { error } = await FirestoreCreateDocument(
+      collectionName,
+      documentId,
+      document,
+    );
+    if (error) {
+      console.log('Error handleCreateUserAuthentification');
+      return;
+    }
+    reset();
+    setContenuModal('init');
+  };
+
+  /**
+   * 2. Envoi des data a Firebase
+   * await le retour du Firebase avec les infos users
+   * Copie des infos users utilent dans un objet
+   * Envoi de l'objet dans la db Firestore
+   */
+
+  const handleCreateUserAuth = async ({
+    email,
+    password,
+    login,
+  }: LoginFormTypeInscription) => {
+    const { error, data } = await firebaseCreateUser(email, password);
+    if (error) {
+      console.log('Error ModalAuthInscription');
+      return;
+    }
+    const userDocumentData = {
+      login: login,
+      uid: data.uid,
+      inscription: new Date(),
+      photoURL: '',
+    };
+    handleCreateUserAuthentification('users', data.uid, userDocumentData);
+  };
+
+  /**
+   * 1. Submit du Form
+   * Test si password > 7 caracteres
+   * Appel function Firebase
+   */
+  const onSubmit = async (data: LoginFormTypeInscription) => {
+    const { password } = data;
+
+    if (password.length < 7) {
+      setError('password', {
+        type: 'manuel',
+        message: 'Le mot de passe doit comporter 7 caractères minimum',
+      });
+      return;
+    }
+    handleCreateUserAuth(data);
+  };
+
+  return (
+    <>
```

Comment:
Message de Sonarlint : A fragment with only one child is redundant.sonarlint(typescript:S6749)

Reply:
corrigé